hypervisor/dm/io_req.c

```diff
@@ -184,7 +186,11 @@ void set_vhm_req_state(struct acrn_vm *vm, uint16_t vhm_req_id, uint32_t state)
 	if (req_buf != NULL) {
 		stac();
 		vhm_req = &req_buf->req_queue[vhm_req_id];
-		atomic_store32(&vhm_req->processed, state);
+		/*
+		 * HV will only set processed to REQ_STATE_PENDING or REQ_STATE_FREE.
+		 * SOS/DM will poll this state whether it's REQ_STATE_PENDING before processing the vhm_req.
```

Comment:
Didn't fix my concern in email

Reply:
I through I answered your comments.
The original implement doesn't consider this because it think wrong that the atomic_store API will operate atomically and imply a memory barrier.  I think the new comments "SOS/DM will poll this state whether it's REQ_STATE_PENDING before processing the vhm_req." may answer your question.

Reply:
@dongyaozu  Comments updated. Could you please help to review this patch again ?